### ISSUE TITLE
Support new TypeScript syntax - get/set in non-class scopes

### ISSUE THREAD

**Issue by RyanCavanaugh** [babel/babel#13071]
## Feature Request

TypeScript 4.3 intends to ship support for different get/set read/write types; this feature requires that we allow new syntax in interfaces and object type literals.

- [ ] I would like to work on this feature!

**Is your feature request related to a problem?**

These are legal in `typescript@next` but not yet supported by Babel
```ts
interface X {
  get foo(): string;
  set foo(v: string | number);
}
type Y = {
  get bar(): string;
  set bar(v: string | number);
}
```

**Describe the solution you'd like**

These just need to get discarded like other TS constructs, natrually.

**Documentation, Adoption, Migration Strategy**

Full details and further discussion available at at https://github.com/microsoft/TypeScript/issues/43420

**Comment by sosukesuzuki** [comment]
I'll work on this

**Comment by nicolo-ribaudo** [comment]
For TS 4.3, we also need to add support for the `override` modifier in class elements and for static index signatures

https://devblogs.microsoft.com/typescript/announcing-typescript-4-3-beta/

**Comment by existentialism** [comment]
@nicolo-ribaudo I'll work on `override`

**Comment by JLHwung** [comment]
This issue has been fixed in #13089
